src/main/java/org/testng/internal/Invoker.java

```diff
@@ -736,7 +732,7 @@ private void invokeBeforeGroupsConfigurations(ITestNGMethod currentTestMethod,
       }
     }
 
-    ITestNGMethod[] beforeMethodsArray = filteredMethods.toArray(new ITestNGMethod[filteredMethods.size()]);
+    ITestNGMethod[] beforeMethodsArray = filteredMethods.toArray(new ITestNGMethod[0]);
```

Comment:
Why this change? It has lower performance.

Reply:
@juherr - No not on JDK8. From what I read, this was a problem on JDK6. This part has been optimized in the newer JDKs.
As per IntelliJ documentation, working with predefined array sizes can cause data races. That is why I changed this.

Reply:
Ok, I've never heard about it! 👍

Reply:
Ditto, didn't know about that. Thanks @krmahadevan!